Update installation instructions adding uv and pixi (#6539)
gammapy/gammapy: docs/getting-started/install.rst

Reviewer: ```suggestion

For example:
```

Reviewer: Should we reference the above sections here too then? i.e. link to the pip and conda sections

Author: done

============

There are numerous ways to install Python and Gammapy as a user. On this page, we list the most common ones.
In general, **we recommend using** :ref:`virtual environments <virtual-envs>` or
:ref:`project-based installation <install-project-based>` when using Gammapy.

> Specially with the latter, you have complete control over the additional packages that you may use in your analysis and you work with

Reviewer: ```suggestion
Specifically with the latter, you have complete control over the additional packages that you may use in your analysis and you work with
```

.. code-block:: bash

>     conda install -c conda-forge gammapy=2.1

Reviewer: ```suggestion
    conda install -c conda-forge gammapy=2.0
```

We should leave this as an LTS

.. code-block:: bash

>     python -m pip install gammapy==2.1

Reviewer: ```suggestion
    python -m pip install gammapy==2.0
```

.. warning::
   Note that environment-based installs do not automatically document your full dependency state,
   which can make analyses harder to reproduce later. For full reproducibility, we recommend
   :ref:`install-project-based` installations.

.. _install-project-based:

Project-based
-------------

Project-based environments are the best way to manage your analysis projects.

* Faster installation method than pip and conda/mamba.
* They create an isolated environment that is tightly coupled to a single analysis project.
* Fully reproducible environments and analysis thanks to a lock file that pins every dependency
  to exact versions, so anyone cloning the project gets an identical environment.
* Makes projects easy to share and set up consistently across systems.
* Follows best practices for reproducible science.

Using uv
^^^^^^^^
`uv`_ is a modern Python package manager that provides a
simple and efficient way to create and manage project-based environments (only with Python
dependencies).

1. Install **uv** following the `uv installation instructions`_.

2. Create a new project in a new directory, by executing the following in a terminal:

   .. code-block:: bash

>       uv init my-science-project

Reviewer: We could change the name as we are talking about gammapy, `gammapy-project` ?

Author: `gammapy-project` sounds to me like you are getting the gammapy repository or code base. Here I wanted to emphasize that you will work on an analysis-case project. But okay, the name is just an example.

Author: What about `my-gammapy-analysis` same as it is used for venvs?

      cd my-science-project

3. Then, to install Gammapy, execute this command in a terminal:

   .. code-block:: bash

      uv add gammapy

This will create a set of files in the new directory *under version control*, including a ``pyproject.toml``
file and a ``uv.lock`` file that contains the exact versions of all used dependencies. Further dependencies
can added be via ``uv add`` command again and they will be added to the lock file. To update Gammapy, execute
``uv add gammapy`` again and it will update to the latest version. You can also install a specific version
like, for example, ``uv add gammapy==2.1``.

Using pixi
^^^^^^^^^^
`pixi`_ is a similar package manager that also provides project-based
environments in a fast and reproducible way. The main difference is that pixi also supports non-Python
dependencies, which can be useful for some users. With **pixi**, you can also define tasks that can be executed
in the environment.

1. Install **pixi** following the `pixi installation instructions`_.

2. Create a new project in a new directory, by executing the following in a terminal:

   .. code-block:: bash

>       pixi init my-second-science-project

Reviewer: Again, as above. Thoughts?

      cd my-second-science-project

3. Then, to install Gammapy, execute this command in a terminal:

   .. code-block:: bash

      pixi add gammapy

Find more details on setting up a workspace in the `pixi documentation`_.

.. _install-other:

Run your analysis in the project-based environment
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. note::
   All the following commands are similar if you use **pixi** instead of **uv**.

> Inside the analysis directory created before you can run your analysis scripts as:

Reviewer: ```suggestion
Inside the `uv` directory created previously, you can run a Python analysis script through:
```

.. code-block:: bash

   uv run python my-analysis-script.py

> Working with jupyter notebooks

Reviewer: ```suggestion
Working with Jupyter notebook
```

""""""""""""""""""""""""""""""

> First install jupyterlab:

Reviewer: You could add a link to jupyterlab. 
```suggestion
First install jupyterlab inside your `uv` project:
```
